### PR TITLE
Realtime doc

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

File: docs/realtime.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+### `@event` jobs trigger
+
+Some event will trigger the activation of a job (ie. When a photo has been uploaded, generate thumbnails and extract EXIF metadatas). This should be done as soon as possible after the events
+
+### Sharing ?
+
+While not absolutely necessary, having cozy A notify cozy B when a shared document is changed allows for both better user experience (faster propagation) and better performance (no need to poll every X minutes, the N cozy we are sharing from).
+
+### Logging ?
```
Comment (nono) [inline docs/realtime.md]: What is the link between the continuous changes feed of Couchdb and logging?

File: docs/realtime.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+Origin: http://calendar.mycozy.example.com
+Authorization: Bearer xxAppOrAuthTokenxx=
+Sec-WebSocket-Key: x3JrandomLkh9GBhXDw==
+Sec-WebSocket-Protocol: io.cozy.websocket
+Sec-WebSocket-Version: 1
+```
+
+Then messages are sent using json
+```
+client > {"mehod": "SUBSCRIBE",
```
Comment (nono) [inline docs/realtime.md]: s/mehod/method/g

File: docs/realtime.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+### SUBSCRIBE
+
+A client can send a SUBSCRIBE request to be notified of changes.
+The payload is a selector for the events it wishes to receive
+For now the only possible selector is on type.
+
+`{"mehod": "SUBSCRIBE", payload: {type:"[desired doctype]"}`
+
+If the client wants to receive full documents with each events, it can include an `include_docs:true` parameter in the payload.
```
Comment (nono) [inline docs/realtime.md]: How do we handle the permissions with the real-time? The payload for subscribe should match a permission rule?

File: docs/realtime.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+### Definitions
+
+- **Events feed:** the feed of occurring events. Most of them will come from couchdb, some jobs might also trigger them. There is two types of feeds:
+  - **continuous** allows to follow events as they occurs
+  - **interval** allow the see the history of the feed from any given time
+
+
+- **Realtime:** user experienced updates of the interface from change happening from another source. **Ie,** *I have a folder opened in cozy-files on the browser, I take some pictures from my smartphone, the pictures appears in the folder without me needing to refresh the browser tab.*
+
+### What couchdb offers
```
Comment (nono) [inline docs/realtime.md]: We also have events that are not coming from CouchDB:

- when the user closes its session
- the jobs system

File: docs/realtime.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+- **SockJS & cie** they are **a lot** of packages which imitate Websocket API while using complicated client&server polyfill to allow support of older browser. [SockJS](https://github.com/sockjs/) is a drop-in websocket replacement with a go package and javascript client.
+
+### Choice = Websocket
+
+While SSE appears at first glance like a better fit for our use case, its limitation and lack of browser priority makes us choose websocket. In the event older browser supports becomes necessary we can use SockJS.
+
+### optimization paths (future)
+
+- **bandwith** Limiting the number of events sent by allowing the client to specified it is only interested in events matching a selector *(files app only care about changes in the files of the current folder view)*
```
Comment (nono) [inline docs/realtime.md]: s/bandwith/bandwidth/

File: docs/realtime.md (hunk 1 above)
Comment (aenario) [inline docs/realtime.md]: The idea is that we will probably have a log.Debug everywhere we have a realtime.Event, but it's not really relevant. Removing it.

**Pull request merged**